### PR TITLE
fix(install packages): retry install when fail

### PR DESCRIPTION
**Pull request by juliayakovlev**
Packages installation fail from time to time during node setup.
Decided to add retrying for that.

Refs: https://github.com/scylladb/scylla-cluster-tests/issues/6317

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2055,7 +2063,9 @@ def offline_install_scylla(self, unified_package, nonroot):
             self.remoter.sudo(f'yum install -y {additional_pkgs}')
             self.remoter.sudo('amazon-linux-extras install java-openjdk11')
         elif self.distro.is_rhel_like:
-            self.remoter.run(f'sudo yum install -y java-11-openjdk-headless {additional_pkgs}')
+            run_command_with_retry(
```
Comment (fruch) [inline sdcm/cluster.py:2066]: why can't we convert this to use `install_package`

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2041,6 +2042,13 @@ def offline_install_scylla(self, unified_package, nonroot):
         """
         Offline install scylla by unified package.
         """
+        @retrying(n=30, sleep_time=15, allowed_exceptions=(UnexpectedExit, Libssh2_UnexpectedExit,))
+        def run_command_with_retry(command, sudo):
```
Comment (fruch) [inline sdcm/cluster.py:2046]: it's weird we need this, doesn't our run command has retries parameter on it's own ?

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1767,7 +1768,7 @@ def download_scylla_manager_repo(self, scylla_repo: str) -> None:
                                   f"--keyserver hkp://keyserver.ubuntu.com:80 --recv-keys {apt_key}", retry=3)
             self.remoter.sudo("apt-get update", ignore_status=True)
 
-    @retrying(n=30, sleep_time=15, allowed_exceptions=UnexpectedExit)
+    @retrying(n=30, sleep_time=15, allowed_exceptions=(UnexpectedExit, Libssh2_UnexpectedExit,))
```
Comment (fruch) [inline sdcm/cluster.py:1771]: this is a good catch... that we are not retry correctly

File: sdcm/cluster.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/cluster.py:2046]: Yes, you are right. There is `retry` parameter of function

File: sdcm/cluster.py [hunk 4]
```diff
@@ -2055,7 +2056,8 @@ def offline_install_scylla(self, unified_package, nonroot):
             self.remoter.sudo(f'yum install -y {additional_pkgs}')
             self.remoter.sudo('amazon-linux-extras install java-openjdk11')
         elif self.distro.is_rhel_like:
-            self.remoter.run(f'sudo yum install -y java-11-openjdk-headless {additional_pkgs}')
+            self.install_package(package_name=f'java-11-openjdk-headless {additional_pkgs}')
+            # self.remoter.run(f'sudo yum install -y java-11-openjdk-headless {additional_pkgs}', retry=10)
```
Comment (fruch) [inline sdcm/cluster.py:2060]: remove comment, once you've proven this is working as expected

**Comment by juliayakovlev** [comment]
artifacts-oel76-test was run 25 times and passed:
https://jenkins.scylladb.com/job/scylla-staging/job/yulia/job/artifacts-oel76-test/

**Comment by juliayakovlev** [comment]
Running [perf-regression-latency-250gb-with-nemesis](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/yulia/job/scylla-perf-regression-latency-250gb-with-nemesis/4/)

**Comment by fruch** [comment]
> Running [perf-regression-latency-250gb-with-nemesis](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/yulia/job/scylla-perf-regression-latency-250gb-with-nemesis/4/)

why ? how it's related to this PR ?

**Comment by juliayakovlev** [comment]
> > Running [perf-regression-latency-250gb-with-nemesis](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/yulia/job/scylla-perf-regression-latency-250gb-with-nemesis/4/)
> 
> why ? how it's related to this PR ?

Originally this issue was opened due to problem in "perf-latency-with-nemesis" job

**Comment by juliayakovlev** [comment]
> > > > Running [perf-regression-latency-250gb-with-nemesis](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/yulia/job/scylla-perf-regression-latency-250gb-with-nemesis/4/)
> > > 
> > > 
> > > why ? how it's related to this PR ?
> > 
> > 
> > Originally this issue was opened due to problem in "perf-latency-with-nemesis" job
> 
> I don't think the pref are relevant for those fixes

@fruch found the issue happened in "perf-latency-with-nemesis" when running `sudo yum install -y wget` on loader.
So I ran similar test

**Comment by juliayakovlev** [comment]
> Running [perf-regression-latency-250gb-with-nemesis](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/yulia/job/scylla-perf-regression-latency-250gb-with-nemesis/4/)

Test passed

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**